File: CSharp/Yow.CoD.Finance.Domain/Contracts/LoanCustomerContactChangedEvent.cs
namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class LoanCustomerContactChangedEvent : Event
    {
        public LoanCustomerContactChangedEvent(string customerContactName, string customerContactPreferredPhoneNumber, string customerContactAlternatePhoneNumber, string customerContactPostalAddress)
        {
            CustomerContact = new CustomerContact(customerContactName, customerContactPreferredPhoneNumber, customerContactAlternatePhoneNumber, customerContactPostalAddress);
        }

        public CustomerContact CustomerContact { get; }
    }
}

File: java/src/main/java/com/leecampbell/cod/domain/contracts/LoanCustomerContactChangedEvent.java
package com.leecampbell.cod.domain.contracts;

public final class LoanCustomerContactChangedEvent extends DomainEvent
{
    private final CustomerContact customerContact;

    public LoanCustomerContactChangedEvent(String customerContactName, 
            String customerContactPreferredPhoneNumber,
            String customerContactAlternatePhoneNumber, 
            String customerContactPostalAddress) {
        
        this.customerContact = new CustomerContact(
            customerContactName, 
            customerContactPreferredPhoneNumber,
            customerContactAlternatePhoneNumber, 
            customerContactPostalAddress);
    }

    public CustomerContact customerContact() {
        return customerContact;
    }
}